packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -775,31 +775,69 @@ contract Deploy is Deployer {
         console.log("Absolute prestate: %s", vm.toString(mipsAbsolutePrestate));
 
         DisputeGameFactory factory = DisputeGameFactory(mustGetAddress("DisputeGameFactoryProxy"));
-        for (uint8 i; i < 2; i++) {
-            Claim absolutePrestate =
-                Claim.wrap(i == 0 ? bytes32(cfg.faultGameAbsolutePrestate()) : mipsAbsolutePrestate);
-            IBigStepper faultVm =
-                IBigStepper(i == 0 ? address(new AlphabetVM(absolutePrestate)) : mustGetAddress("Mips"));
-            GameType gameType = GameType.wrap(i);
-            if (address(factory.gameImpls(gameType)) == address(0)) {
-                factory.setImplementation(
-                    gameType,
-                    new FaultDisputeGame({
-                    _gameType: gameType,
-                    _absolutePrestate: absolutePrestate,
-                    _maxGameDepth: i == 0 ? 4 : cfg.faultGameMaxDepth(), // The max depth of the alphabet game is always 4
+
+        // -------------
+        // Mainnet games
+        // TODO: Remove the `onlyDevnet` modifier and conditionally deploy the mainnet games based on the chainid.
+        // -------------
+
+        // Set the Cannon FaultDisputeGame implementation in the factory.
+        _setFaultGameImplementation(
+            factory,
+            GameTypes.FAULT,
+            Claim.wrap(mipsAbsolutePrestate),
+            IBigStepper(mustGetAddress("Mips")),
+            cfg.faultGameMaxDepth()
+        );
+
+        // -------------
+        // Devnet games
+        // Devnet-specific games are created with a game type descending from the `GameType` type's max value.
+        // This is to preserve the game type space for mainnet games and to have those be in parity with
+        // their prod counter parts when deployed to a devnet.
+        // -------------
+
+        // Set the Alphabet FaultDisputeGame implementation in the factory.
+        Claim alphabetAbsolutePrestate = Claim.wrap(bytes32(cfg.faultGameAbsolutePrestate()));
+        _setFaultGameImplementation(
+            factory,
+            GameType.wrap(255),
+            alphabetAbsolutePrestate,
+            IBigStepper(new AlphabetVM(alphabetAbsolutePrestate)),
+            4 // The max game depth of the alphabet game is always 4.
+        );
```

Comment:
These are now good candidates for their own functions. Especially since the parameters are entirely mutually exclusive less the factory.

The devnet function can then also verify that the chain id is indeed the devnet.

Reply:
I think we should keep it how it is now due to us not wanting the deploy script to deploy the fault proof contracts to live networks yet, but agreed that these should be split out soon.

After tower defense, will be nice to re work the deployment of these contracts :)

Reply:
I think we said the same after bedrock migration 🤣  But sure, nbd